### PR TITLE
Serverless/Use case Queries

### PR DESCRIPTION
**Pull request by Awsaf780**
Fixes #

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [x] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
Query data from Frontend

## What was done
- Added GET query events for Use cases
- Added a Single function to Handle all queries
- Added permission to Query from Dynamo

## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: awsaf-serverless-engine/handler.js [hunk 1]
```diff
@@ -169,44 +188,132 @@ module.exports.hello = async (event) => {
...
+            }
+          } else {
+            return sendResponseToClient(400, JSON.stringify("Query Must contain Customer Property in Data"));
+          }
+        }
 
-    const url = await s3.getSignedUrl("putObject", signedParams);
+        else if (query.queryType === 'queryCustomerWithDate') {
+          if (query.data?.customer && query.data?.initialDate && query.data?.finalDate) {
+            const indexName = "gsi1-pk-sk-index";
```
[the remark this answers is not in the document]
Comment (Awsaf780) [reply awsaf-serverless-engine/handler.js]: Resolved in commit [c09a41d](https://github.com/CraftsmenLtd/coe-batch-q322/pull/287/commits/c09a41d9128c3480d5efb972cda7e531a021867a)

File: awsaf-serverless-engine/handler.js [hunk 2]
```diff
@@ -169,44 +188,132 @@ module.exports.hello = async (event) => {
...
+            const keyConditionExpression = 'gsi1pk = :gsi1pk and gsi1sk BETWEEN :i_date AND :f_date';
+            const expressionAttributeValues = {
+              ':gsi1pk': "USER_ACTIVITY",
```
[the remark this answers is not in the document]
Comment (Awsaf780) [reply awsaf-serverless-engine/handler.js]: This was very helpful. Resolved in commit [c09a41d](https://github.com/CraftsmenLtd/coe-batch-q322/pull/287/commits/c09a41d9128c3480d5efb972cda7e531a021867a)

File: awsaf-serverless-engine/handler.js [hunk 3]
```diff
@@ -169,44 +188,132 @@ module.exports.hello = async (event) => {
...
+          } else {
+            return sendResponseToClient(400, JSON.stringify("Query Must contain Customer Property with Date Range"));
+          }
+        }
+
+        else if (query.queryType === 'queryUser') {
+          if (query.data?.customer && query.data?.user) {
+            const keyConditionExpression = 'pk = :pk and begins_with(sk, :sk)';
+            const expressionAttributeValues = {
+                  ':pk': `CUSTOMER_ACTIVITY_${query.data.customer}`,
```
[the remark this answers is not in the document]
Comment (Awsaf780) [reply awsaf-serverless-engine/handler.js]: Resolved in commit [c09a41d](https://github.com/CraftsmenLtd/coe-batch-q322/pull/287/commits/c09a41d9128c3480d5efb972cda7e531a021867a)

File: awsaf-serverless-engine/handler.js [hunk 4]
```diff
@@ -169,44 +188,132 @@ module.exports.hello = async (event) => {
...
 
-    const response = {
-      statusCode: 200,
-      body: url,
-    };
+        else if (query.queryType === 'queryUserWithDate') {
+          if (query.data?.user && query.data?.initialDate && query.data?.finalDate) {
+            const keyConditionExpression = 'pk = :pk and sk BETWEEN :i_date AND :f_date';
+            const expressionAttributeValues = {
+              ':pk': `CUSTOMER_ACTIVITY_${query?.data.customer}`,
```
[the remark this answers is not in the document]
Comment (Awsaf780) [reply awsaf-serverless-engine/handler.js]: Resolved in commit [c09a41d](https://github.com/CraftsmenLtd/coe-batch-q322/pull/287/commits/c09a41d9128c3480d5efb972cda7e531a021867a)

**Review by mhazad — APPROVED** [review]
LGTM

**Pull request merged**